moped-editor/src/views/projects/projectView/ProjectComponentsMapEdit.js

```diff
@@ -257,9 +256,11 @@ const ProjectComponentsMapEdit = ({
    * @param {String} newValue - The new value from the autocomplete selector
    */
   const handleComponentSubtypeSelect = (e, newValue) => {
-    setSelectedComponentSubtype(
-      (newValue ?? e.target.value ?? "").toLowerCase()
-    );
+    const componentName =
+      selectedComponentType.toLowerCase().replaceAll(" ", "") +
+      newValue.toLowerCase().replaceAll(" ", "");
```

Comment:
Thank you for explaining to me the reasoning behind doing this comparison using the combined names as a key to lookup instead of pushing this back to the database to resolve.